### PR TITLE
WIP (CPF-130) Monitor and Alarming

### PR DESCRIPTION
**Pull request by jlhughes**
Initial Monitor and Alarm models.
Plus updates to configure via EasyDAO.

### CODE REVIEW

File: src/foam/dao/EasyDAO.js [hunk 1]
```diff
@@ -343,6 +389,43 @@ return delegate;
...
+      name: 'ruler',
+      class: 'Boolean',
+      value: false
+    },
+    {
+      name: 'rulerDaoKey',
+      class: 'String',
+      factory: function() {
+        return this.serviceName;
+      }
```
Comment (lchanmann) [inline src/foam/dao/EasyDAO.js]: factory doesn't generate Java code.

File: src/foam/dao/EasyDAO.js [hunk 2]
```diff
@@ -332,6 +374,10 @@ return delegate;
     {
       /** Simpler alternative than providing serverBox. */
       name: 'serviceName',
+      class: 'String',
+      factory: function() {
+        return this.nSpec && this.nSpec.name;
+      },
       generateJava: false
```
Comment (lchanmann) [inline src/foam/dao/EasyDAO.js]: we need Java code generation.

File: src/foam/nanos/monitor/test/TestMonitor.js [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+      Alarm alarm = monitor.findAlarm(getX());
+      if ( alarm == null ) {
+        alarm = new Alarm.Builder(getX()).setName("OK").build();
+      } else {
+        alarm = (Alarm) alarm.fclone();
+      }
+      alarm.setRepeated(alarm.getRepeated() + 1);
+      alarm = (Alarm) monitor.getAlarms(getX()).put(alarm);
+      monitor.setAlarm(alarm.getId());
+      ((foam.dao.DAO) getX().get("monitorDAO")).put(monitor);
```
Comment (lchanmann) [inline src/foam/nanos/monitor/test/TestMonitor.js]: No need to `put(monitor)` RulerDAO will take care of it.

**Review by lchanmann — CHANGES_REQUESTED** [review]

**Comment by jlhughes** [comment]
Really messed up after experimenting with cleaning up files.

**Pull request closed without merging**